debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BlockingReader.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import org.apache.kafka.connect.source.SourceRecord;
+
+/**
+ * A component that blocks doing nothing until the connector task is stopped
+ * 
+ * @author Peter Goransson
+ *
+ */
+public class BlockingReader extends AbstractReader {
+
+    private final CountDownLatch latch = new CountDownLatch(1);
+    
+    public BlockingReader(String name, MySqlTaskContext context) {
+        super(name, context);
+    }
+
+    @Override
+    protected void doStart() {
+        logger.info("Connector has completed all of its work but will continue in the running state. It can be shutdown at any time.");
+    }
+
+    @Override
+    protected void doStop() {
+        this.completeSuccessfully();
+        latch.countDown();
+    }
+
+    @Override
+    protected void doCleanup() {
+        logger.debug("Blocking Reader has completed.");
+    }
+    
+    /**
+     * Waits indefinitely until the connector task is shut down
+     */
+    @Override
+    public List<SourceRecord> poll() throws InterruptedException {                
+        latch.await();   
+        return super.poll();
+    }
+
```

Comment:
`AbstractReader` seems to be fairly heavy weight. Did you consider having this just implement `Reader`, and if so what were the tradeoffs that made you go this route?

(Really, I'm just asking. I haven't thought much about this and am interested in your thoughts.)

Reply:
When I tried just implementing Reader it ran into some issues. One example: it appeared to Re-Snapshot the DB every time the connector restarted.

So I figured extending AbstractReader was a requirement. Maybe it's not I'm just new to the codebase so didn't dig too deep.